pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,290 @@
+        self.basedir = self.cfg.getpath(cfgsect, 'basedir', '.', abs=True)
+        self.nout = {}
+        for sname in self._scene_cfg:
+            name = self._scene_cfg[sname]['image_name']
+            self.nout[name] = file_index(self.basedir, name) if intg.isrestart else 0
+
+        # Set subdivision
+        self.divisor = self.cfg.getint(cfgsect, 'devision', 
+                                       self.cfg.getint('solver', 'order'))
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Share number of ele types per part
+        ndoms = len(system.ele_map)
+        ndoms = comm.gather(ndoms, root=root)
+        if rank == root:
+            ndoms = np.asarray(ndoms)
+        ndoms = comm.bcast(ndoms, root=root)
+        self.dom_0 = dom_0 = np.sum(ndoms[:rank])
+
+        # Generate conduit blueprint of mesh
+        mesh_n = ConduitNode()
+        self.ops = {}
+        for i, (ename, eles) in enumerate(system.ele_map.items(), start=dom_0):
+            if ename in ['pri', 'pyr']:
+                raise ValueError(f'Ascent: {ename} elements not supported')
+
+            dpf = f'domain_{i}/'
+
+            svpts = self._get_std_ele(ename, eles.nspts)
+            nsvpts = len(svpts)
+
+            mesh_op = self._get_mesh_op(ename, eles.nspts, svpts)
+            soln_op = self._get_soln_op(ename, eles.nspts, svpts)
+            self.ops[ename] = (mesh_op, soln_op)
+
+            mesh_n[dpf + 'state/domain_id'] = rank
+            mesh_n[dpf + 'coordsets/coords/type'] = 'explicit'
+            mesh_n[dpf + 'topologies/mesh/coordset'] = 'coords'
+            mesh_n[dpf + 'topologies/mesh/type'] = 'unstructured'
+            mesh_n[dpf + 'topologies/mesh/elements/shape'] = ename
+
+            subdvcls = subclass_where(BaseShapeSubDiv, name=ename)
+            sconn = subdvcls.subnodes(self.divisor)
+            subdvcon = np.hstack([sconn + j*nsvpts for j in range(eles.neles)])
+            mesh_n[dpf + 'topologies/mesh/elements/connectivity'] = subdvcon
+
+            xd = np.einsum('ik,k...', mesh_op, eles.ploc_at_np('linspts'))
+            X = xd.reshape(eles.ndims, -1)
+            for j, x in enumerate('xyz'[:eles.ndims]):
+                mesh_n[dpf + f"coordsets/coords/values/{x}"] = X[j,:]
+
+            for field in self._expr:
+                mesh_n[dpf + f'fields/{field}/association'] = 'vertex'
+                mesh_n[dpf + f'fields/{field}/volume_dependent'] = 'false'
+                mesh_n[dpf + f'fields/{field}/topology'] = 'mesh'
+
+        self.mesh_n = mesh_n
+
+        # Inital Ascent setup
+        self.a = Ascent()
+        self.ascent_config = ConduitNode()
+        self.ascent_config['mpi_comm'] = comm.py2f()
+        self.ascent_config['runtime/type'] = 'ascent'
+        self.ascent_config['runtine/vtkm/backend'] = 'serial'
+
+        # Open ascent
+        self.a.open(self.ascent_config)
+
+    def __del__(self):
+        self.a.close()
```

Comment:
Check the attribute `a` exists first in case an exception is thrown during init.